### PR TITLE
feat: Support for multi-use statements in `NamespaceUsesAnalyzer`

### PR DESCRIPTION
**Pull request by Wirone**
The goal of this refactor/enhancement is being able to collect imports also from comma-separated and `{}`-grouped multi-use statements. I decided to add explicit `isInMulti` flag, even though "multi" flag could be inferred from start-end chunk indices - I thought it's better, because maybe we can pass these chunk indices even for single symbols in the future.

Change is backward compatible because class is marked as internal, and entry-point for collecting imports has opt-in flag, so all the fixers work exactly the same at this point. I'll create separate PRs that will show how this feature is used in practice.

### CODE REVIEW

File: src/Tokenizer/Analyzer/NamespaceUsesAnalyzer.php [hunk 1]
```diff
@@ -57,45 +57,125 @@ public function getDeclarationsInNamespace(Tokens $tokens, NamespaceAnalysis $na
...
+                    $imports[] = new NamespaceUseAnalysis(
+                        $type,
+                        $qualifiedName['fullName'],
+                        $qualifiedName['shortName'],
+                        $qualifiedName['aliased'],
+                        $multi,
+                        $startIndex,
+                        $endIndex,
+                        $multi ? $index : null,
+                        $multi ? $tokens->getPrevMeaningfulToken($qualifiedName['afterIndex']) : null
```
Comment (julienfalque) [inline src/Tokenizer/Analyzer/NamespaceUsesAnalyzer.php:137]:
```suggestion
                        $multi ? $previousToken : null
```

File: src/Tokenizer/Analyzer/NamespaceUsesAnalyzer.php [hunk 2]
```diff
@@ -57,45 +57,125 @@ public function getDeclarationsInNamespace(Tokens $tokens, NamespaceAnalysis $na
...
+                    );
+                }
 
-            if ($token->isGivenKind(CT::T_FUNCTION_IMPORT)) {
-                $type = NamespaceUseAnalysis::TYPE_FUNCTION;
-            } elseif ($token->isGivenKind(CT::T_CONST_IMPORT)) {
-                $type = NamespaceUseAnalysis::TYPE_CONSTANT;
+                $index = $qualifiedName['afterIndex'];
+                $aliased = false;
+                $fullName = $shortName = '';
```
Comment (julienfalque) [inline src/Tokenizer/Analyzer/NamespaceUsesAnalyzer.php:143]: I think these variables are unused leftovers of the refactoring to create `getNearestQualifiedName()`.

File: src/Tokenizer/Analyzer/NamespaceUsesAnalyzer.php [hunk 3]
```diff
@@ -106,16 +186,24 @@ private function parseDeclaration(Tokens $tokens, int $startIndex, int $endIndex
...
-        return new NamespaceUseAnalysis(
-            trim($fullName),
-            $shortName,
-            $aliased,
-            $startIndex,
-            $endIndex,
-            $type
-        );
+        return [
+            'fullName' => trim($fullName),
```
[the remark this answers is not in the document]
Comment (Wirone) [reply src/Tokenizer/Analyzer/NamespaceUsesAnalyzer.php:203]: It was there before, I did not analyse it. But I'll check 🙂.

File: src/Tokenizer/Analyzer/NamespaceUsesAnalyzer.php [hunk 4]
```diff
@@ -28,22 +28,22 @@ final class NamespaceUsesAnalyzer
     /**
      * @return list<NamespaceUseAnalysis>
      */
-    public function getDeclarationsFromTokens(Tokens $tokens): array
+    public function getDeclarationsFromTokens(Tokens $tokens, bool $allowMultiUses = false): array
```
[the remark this answers is not in the document]
Comment (Wirone) [reply src/Tokenizer/Analyzer/NamespaceUsesAnalyzer.php:31]: Yeah, probably we should. All places where analyser is used must be verified and obviously I did not want to do it in single PR 😅. So this is basically a migration step.

**Review by julienfalque — APPROVED** [review]

**Pull request merged**